File: .tmp_sc4013/java_06/2288_apply_discount_to_prices.java
// LeetCode 2288 - Apply Discount to Prices
// https://leetcode.com/problems/apply-discount-to-prices/

class Solution {
    public String discountPrices(String sentence, int discount) {
        String[] parts = sentence.split(" ");
        for (int i = 0; i < parts.length; i++) {
            String part = parts[i];
            if (part.length() >= 2 && part.charAt(0) == '$') {
                boolean ok = true;
                for (int j = 1; j < part.length(); j++)
                    if (part.charAt(j) < '0' || part.charAt(j) > '9') {
                        ok = false;
                        break;
                    }
                if (ok) {
                    long val = Long.parseLong(part.substring(1));
                    double price = val * (100.0 - discount) / 100.0;
                    parts[i] = String.format("$%.2f", price);
                }
            }
        }
        return String.join(" ", parts);
    }
}

File: .tmp_py4013/batch_02_sources/2288_apply_discount_to_prices.js
// LeetCode 2288 - Apply Discount to Prices
// https://leetcode.com/problems/apply-discount-to-prices/

var discountPrices = function(sentence, discount) {
    const parts = sentence.split(' ');
    for (let i = 0; i < parts.length; i++) {
        const part = parts[i];
        if (part.length >= 2 && part[0] === '$') {
            let ok = true;
            for (let j = 1; j < part.length; j++) {
                if (part[j] < '0' || part[j] > '9') { ok = false; break; }
            }
            if (ok) {
                const val = Number(part.slice(1));
                const price = val * (100 - discount) / 100;
                parts[i] = '$' + price.toFixed(2);
            }
        }
    }
    return parts.join(' ');
};
